lmnet/lmnet/datasets/bdd100k.py

```diff
@@ -74,41 +94,54 @@ def __init__(self,
+        self.img_dir = os.path.join(self.data_dir, "images", "100k", subset_dir)
+        self.anno_dir = os.path.join(self.data_dir, "labels", "bdd100k_labels_images_" + subset_dir + ".json")
         self.paths = []
         self.bboxs = []
+        self.save_file = os.path.join("dataset", "bdd100k_%s.pickle" % self.subset)
+        if not os.path.exists(os.path.dirname(self.save_file)):
+            os.makedirs(os.path.dirname(self.save_file))
 
         self._init_files_and_annotations()
 
     def _init_files_and_annotations(self):
-        img_paths = glob.glob(join(self.img_dir, "*.jpg"))
-        anno_paths = [join(self.anno_dir, splitext(basename(f))[0] + ".json")
-                      for f in img_paths]
-
-        self.paths = img_paths
-        for f in anno_paths:
-            with open(f) as fp:
-                raw = json.load(fp)
-            objs = raw["frames"][0]["objects"]
-            bbox = []
-            for obj in objs:
-                cat = obj["category"]
-                cat = cat.replace(" ", "_")
-                if cat in self.classes:
-                    cls_idx = self.classes.index(cat)
-                    x1 = int(round(obj["box2d"]["x1"]))
-                    x2 = int(round(obj["box2d"]["x2"]))
-                    y1 = int(round(obj["box2d"]["y1"]))
-                    y2 = int(round(obj["box2d"]["y2"]))
-                    x = x1
-                    y = y1
-                    w = x2 - x1
-                    h = y2 - y1
-
-                    bbox.append([x, y, w, h, cls_idx])
-            bbox = np.array(bbox, dtype=np.int32)
-            self.bboxs.append(bbox)
+        img_paths = OrderedDict([(os.path.basename(path), path)
+                                 for path in glob.glob(os.path.join(self.img_dir, "*.jpg"))])
+        img_names = set(img_paths.keys())
+
+        anno_data = json.load(open(self.anno_dir))
+        bbox = [[] for _ in range(len(img_paths))]
+        bbox = OrderedDict(zip(list(img_paths.keys()), bbox))
+
+        counts = 0
+        for item in anno_data:
+            counts += 1
+            # Skip if Label not in images
+            if item['name'] not in img_names:
+                continue
+            for label in item['labels']:
+                class_name = label['category'].replace(' ', '_')
+                # Skip if Classname/Category not in Selected classes
+                if class_name not in self.classes:
+                    continue
+
+                cls_idx = self.classes.index(class_name)
+                x1 = int(round(label["box2d"]["x1"]))
+                x2 = int(round(label["box2d"]["x2"]))
+                y1 = int(round(label["box2d"]["y1"]))
+                y2 = int(round(label["box2d"]["y2"]))
+
+                x = x1
+                y = y1
+                w = x2 - x1
+                h = y2 - y1
+
+                index = item['name']
+                bbox[index] += [[x, y, w, h, cls_idx]]
+        print()
```

Comment:
Now, this print function is not necessary. Could you remove this part?

Reply:
Removed